doc/reference/src/low-level-execution-model.pdc

```diff
@@ -111,276 +111,542 @@ The oracle-free abstract algorithm for optimal reduction operates on four node t
+    The $\lambda$ /\ $\mu$ case is special in that in a Valid EAL-Net the node will be traversed via the primary port via the parent AST and then by the second auxiliary port.
+    \\ \\
+    The second auxiliary port refers to the variable binding.
+    \\ \\
+    When we traverse the $\lambda$ node by the primary port first (this must be the case), we add a new var into the first map.
+    \\ \\
+    Then we run the recursive algorithm on Auxiliary 1, this subterm will only access the second Auxiliary port of this $\lambda$ node.
+    \\ \\
+    If the term is unused, then we will connect an eraser node to the second Auxiliary port.
+    \\ \\
+    This thus preserves the number of free ports.
+    \\ \\
+    By induction the subterms on Auxiliary one are also well formed
+    \\ \\
+    $\therefore$ the Lambda term is $α$ equivalent to the normal evaluation answer.
+  \end{tcolorbox}
+
+  \begin{tcolorbox}[title= Case 3 - FanIn, colframe=inner-box-1]
+    FanIn's are another interesting case, for this case we allocate a map from each FanIn to the traversal status.
+    \\ \\
+    For this, we keep a list of the following types
+    \begin{verbatim}
+     type Status = In FanPort
+                 | Complete FanPort
+     type FanPort = Circle | Star
+    \end{verbatim}
+    Where circle is the first auxiliary port and star is the second.
+
+    \begin{figure}[H]
+        \begin{tikzpicture}
+          \inetcell[]{a}{$f_i$}[5]
+          \draw (-0.29, 0.23) node {$\scalebox{1.9}{$\circ$}$};
+          \draw ( 0.29, 0.29) node {$\scalebox{1.7}{$\star$}$};
+          \cutWire{a.out}{b.north}{}{}
+        \end{tikzpicture}
+    \end{figure}
+
+
+    For this analysis we must consider what port we can enter from, however first we should note that before entering the primary port of a FanIn, that we must have first traversed through either $\star$ or $\circ$ first.
+    \\ \\
+    This is the case because entering a FanIn in this way is known as a FanOut, and denotes the end of sharing.
+    And since ending sharing before starting is not possible from a valid EAL-Net, these cases never happen.
+    \\ \\
+    \begin{tcolorbox}[title= Enter from Prim, colframe=inner-box-2]
+      As shown above, we must have already traversed a FanIn node, namely $\star$ or $\circ$.
+      \\ \\
+      This means the map from the FanIn number $i$ is not empty, and thus we have a history of completing nodes or being in nodes.
+      \\ \\
+      If we have an unfinished $In$, then we pick the most recent $In$ port left to traverse.
+      This is forced as this node is considered a FanOut and thus closes sharing.
+      \\ \\
+      We then mark the node as complete. If we have completed a single node, then we simply choose to leave through the other port.
+      \\ \\
+      Since we are working over valid EAL-Net's which require no oracle, there is no ambiguity in this mechanism. Furthermore, this fact also excludes any other configurations from happening.
```

Comment:
How does the fan's label come into play here?

Reply:
"This means the map from the FanIn number $i$ is not empty, and thus we have a history of completing nodes or being in nodes." 

From above, basically we have the label be a map to the status of going through various ports.

Reply:
that's what I thought, we should be clearer about the separation of context by level though